src/generic_sort.cpp

```diff
@@ -373,4 +382,52 @@ std::string GenericDatatypeSort::to_string() const
   return this->compute_string();
 }
 
+DatatypeComponentSort::DatatypeComponentSort(SortKind sk,
+                                             std::string name,
+                                             Sort dt)
+    : GenericSort(sk), name(name), dt_sort(dt)
+{
+  if (sk != CONSTRUCTOR && sk != SELECTOR && sk != TESTER)
+  {
+    throw IncorrectUsageException("Wrong sortkind input");
+  }
+}
+std::string DatatypeComponentSort::compute_string() const { return name; }
+
+std::string DatatypeComponentSort::to_string() const
+{
+  return compute_string();
+}
+
+std::string DatatypeComponentSort::get_uninterpreted_name() const
+{
+  return name;
+}
+
+SortVec DatatypeComponentSort::get_domain_sorts() const
+{
+  std::vector<Sort> domain_sorts;
+  domain_sorts.push_back(dt_sort);
+  return domain_sorts;
+}
```

Comment:
I am not sure if having the datatype as the domain sort is the right thing to do here. 
Instead, I would change `DatatypeComponentSort` to use domain/co-domain as follows :

- Constructor: domain = arguments passed to the constructor, co-domain = datatype
- Tester: domain = datatype, co-domain = bool
- Selector: domain = datatype, co-domain = sort of the selected parameter

What do you think?